doc/style-guide-javascript.md

```diff
@@ -0,0 +1,185 @@
+     *     // Sample usage
+     *     var test;
+     *     test = new FakeTestClass();
+     *     test.run('something');
+     *
+     * This next "global" part flags `window` as a global variable.  That's
+     * because nothing in this first function accesses any globals.
+     */
+    /*global window*/
+    (function (global, wind) {
+        /**
+         * This is a fake class for illustration purposes.
+         *
+         * The class gets a capital letter at the beginning because it is a
+         * constructor.
+         *
+         * @class FakeTestClass
+         */
+        function FakeTestClass() {
+            // In case the user forgot to use `new`, return a new instance
+            if (!(this instanceof FakeTestClass)) {
+                return new FakeTestClass();
+            }
+
+            this.callCount = 0;
+        }
+
+        /**
+         * Sample event handler
+         *
+         * @param {Event} event Unused
+         * @param {AnotherClass~speciallyFormattedObject} data Checks for callCount
+         */
+        FakeTestClass.prototype.onClick = function (event, data) {
+            var lastClickData;
+
+            /**
+             * Copy a property from incoming data into the lastClickData
+             *
+             * This is a function, thus it's split out here and documented.
+             *
+             * @param {string} name
+             */
+            function copyProperty(name) {
+                if (data[name]) {
+                    lastClickData[name] = data[name];
+                }
+            }
+
+            if (typeof data === "object" && data.callCount) {
+                this.callCount = +data.callCount || 0;
+            }
+
+            lastClickData = {};
+            [
+                'mouseX',
+                'mouseY'
+            ].forEach(copyProperty);
+            lastClickData.fakeTestData = {
+                callCount: this.callCount,
+                instance: this
+            };
+            this.lastClickData = lastClickData;
+        }
+
+        /**
+         * Run a test, display a message.
+         *
+         * The method's added to the prototype this way for cleanliness, but
+         * using Object.create() is also an options.  The return value is "this"
+         * so we are not specifying a type - it is possible that the context is
+         * anything and we're returning whatever the context is.
+         *
+         * @param {string} [message] Show this message via an alert
+         * @return this
+         */
+        FakeTestClass.prototype.run = function (message) {
+            if (message) {
+                wind.alert(message);
+            }
+
+            this.callCount += 1;
+        };
+
+        // Export to UMD or the global object - standard boilerplate
+        if (module && module.exports) {
+            module.exports = FakeTestClass;
+        } else {
+            global.FakeTestClass = FakeTestClass;
+        }
+    }(this, window))
```

Comment:
My main point is that we don't have anything about use strict in the guide. I believe it should be in the style guide just to make sure people who reference this, who might be unfamiliar with JS would know we require it and then there could be a bit about if you minify use the function form. Otherwise at the top of the file under the globals definition, if globals is required.